### PR TITLE
Delia/add ssh certificate parser

### PR DESCRIPTION
**Pull request by deliaconstantino**
- migrates `/resources/ssh-certificate-parser.md` to `/next`
- compare to web version at https://goteleport.com/resources/ssh-certificate-parser/
- Preview **[here](https://teleport-docs-next-git-delia-add-ssh-certi-eff78a-gravitational.vercel.app/resources/ssh-certificate-parser/)**

**Notes:**
The existing component on `/web` implements its own [function](https://github.com/gravitational/web/blob/6b85871bbebc34d180ebc2348f04444e4b23abd4/web-assets/js/modules/ssh-cert-parser.js#L52-L56) to decode the html returned in the fetch to a human readable format. Converted this to React with `dangerouslySetInnerHTML`. While this can be dangerous, I assumed that it should be fine in this case because:
a) it was already working in a similar manner on `/web`
b) the server is another part of gravitational, not a third party
https://github.com/gravitational/next/blob/0ae1f5ea07033463d9f492aa802107b063495763/components/SSHParser/SSHParser.tsx#L20-L22

### CODE REVIEW

File: components/SSHParser/SSHParser.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+};
+
+export interface IntructionsProps {
+  handleClick: () => void;
+}
+
+export const Instructions = ({ handleClick }: IntructionsProps) => {
+  return (
+    <>
+      <StyledLabel id="info">INSTRUCTIONS</StyledLabel>
```
Comment (sandylcruz) [inline components/SSHParser/SSHParser.tsx:34]: You added `textTransform: "uppercase"` to `StyledLabel` so this doesn't need to be uppercase

File: pages/resources/ssh-certificate-parser.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import {
+  PageIntro,
+  SSHParser,
+  Section,
+  Centrator,
+  TryTeleport
+} from "components";
+
+<Centrator maxWidth="1240px" padding={["0 16px", "32px 80px 32px 32px"]}>
+  <PageIntro
```
Comment (sandylcruz) [inline pages/resources/ssh-certificate-parser.mdx:21]: `bg` looks like it should be `grayWave`

File: pages/resources/ssh-certificate-parser.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    subtitle="Teleport Access Plane"
+    description="SSH Certificate Parser: Free online decoder for SSH Certificates using Teleport: Modern SSH for Teams Managing Distributed Server Clusters"
+    width="100%"
+    verticalResponsive={false}
+  />
+</Centrator>
+
+<Section bg="flatWhite" width="100%" borderTop="1px solid #F0F2F4">
+  <Centrator maxWidth="1240px"  px={[0, 3]} pb={[0, 8]}>
+    <Section bg="flatWhite" fontColor="darkest" lineHeight="md" fontSize="text-md" padding={["16px", "32px 64px"]}>
```
Comment (sandylcruz) [inline pages/resources/ssh-certificate-parser.mdx:32]: for `padding`, use the theming numbers

Comment (sandylcruz) [reply pages/resources/ssh-certificate-parser.mdx:32]: Also it seems there's too much `padding-bottom` on the `<p>` element

File: components/SSHParser/SSHParser.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                id="ssh"
+                placeholder="Paste your ssh certificate here..."
+                value={formText}
+                onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                  setFormText(e.target.value)
+                }
+                rows={8}
+              />
+            </StyledLabel>
+            <Button
```
Comment (sandylcruz) [inline components/SSHParser/SSHParser.tsx:124]: Needs more space between the text box and `<Button>`

File: components/SSHParser/SSHParser.tsx [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setIsDisabled(false);
+    }
+  }
+
+  return (
+    <Box maxWidth="1240px" width="1240px" padding={[3, "32px 64px"]}>
```
Comment (sandylcruz) [inline components/SSHParser/SSHParser.tsx:103]: Use the theming numbers for `padding`

File: pages/resources/ssh-certificate-parser.mdx (hunk 3 above)
Comment (sandylcruz) [reply pages/resources/ssh-certificate-parser.mdx:32]: `<PageIntro>` should have less `paddingLeft` so the text is in line with `PageIntro`'s `title`

File: components/SSHParser/SSHParser.tsx [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+      <Flex maxWidth="1240px" flexDirection="column">
+        {showCert && <CertCard certInfo={certInfo} />}
+        <Flex
+          flexDirection={["column-reverse", "row"]}
+          alignContent="flex-start"
+        >
+          <StyledForm onSubmit={parseCert}>
+            <StyledLabel id="ssh" color={errorMessage}>
+              {labelMessage}
+              <StyledTextArea
```
Comment (sandylcruz) [inline components/SSHParser/SSHParser.tsx:113]: The `text-area` in the original is taller and wider.

File: components/SSHParser/SSHParser.tsx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+
+export interface CertCardProps {
+  certInfo: string;
+}
+
+export const CertCard = ({ certInfo }: CertCardProps) => {
+  return (
+    <StyledAnimatedBox>
+      <StyledH3>Certificate Info</StyledH3>
+      <Box overflow="auto" fontSize="text-lg" lineHeight="100%">
```
Comment (sandylcruz) [inline components/SSHParser/SSHParser.tsx:20]: `lineHeight` in the original seems to be about `19px`

Comment (sandylcruz) [reply components/SSHParser/SSHParser.tsx:20]: On smaller screens, there needs to `paddingLeft` or `marginLeft`

File: components/SSHParser/SSHParser.tsx [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+export interface IntructionsProps {
+  handleClick: () => void;
+}
+
+export const Instructions = ({ handleClick }: IntructionsProps) => {
+  return (
+    <>
+      <StyledLabel id="info">INSTRUCTIONS</StyledLabel>
+      <StyledOl>
```
Comment (sandylcruz) [inline components/SSHParser/SSHParser.tsx:35]: `marginLeft` should be less.

**Review by sandylcruz — COMMENTED** [review]
Overall, really good! Great work on the parsing! Left a few comments.

File: pages/resources/ssh-certificate-parser.mdx (hunk 2 above)
Comment (deliaconstantino) [reply pages/resources/ssh-certificate-parser.mdx:21]: updated

File: pages/resources/ssh-certificate-parser.mdx (hunk 3 above)
Comment (deliaconstantino) [reply pages/resources/ssh-certificate-parser.mdx:32]: updated these

File: components/SSHParser/SSHParser.tsx (hunk 5 above)
Comment (deliaconstantino) [reply components/SSHParser/SSHParser.tsx:103]: good point, updated these as well

File: components/SSHParser/SSHParser.tsx (hunk 6 above)
Comment (deliaconstantino) [reply components/SSHParser/SSHParser.tsx:113]: thanks, this was missing the border-box

File: pages/resources/ssh-certificate-parser.mdx (hunk 2 above)
Comment (deliaconstantino) [reply pages/resources/ssh-certificate-parser.mdx]: updated this to `grayWave`, though they actually look the same to me. Do you see a difference?

File: components/SSHParser/SSHParser.tsx (hunk 5 above)
Comment (deliaconstantino) [reply components/SSHParser/SSHParser.tsx]: thank you!

File: components/SSHParser/SSHParser.tsx (hunk 6 above)
Comment (deliaconstantino) [reply components/SSHParser/SSHParser.tsx:113]: also increased rows to `rows={9}` to account for the different lineHeights available in the `next` theme

**Review by sandylcruz — APPROVED** [review]

**Pull request merged**